src/API/Site/Controllers/Ads/BudgetMetricsController.php

```diff
@@ -0,0 +1,205 @@
+ * Class BudgetMetricsController
+ *
+ * ContainerAware used for:
+ * - BudgetMetrics
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads
+ */
+class BudgetMetricsController extends BaseController implements ContainerAwareInterface, ISO3166AwareInterface {
+
+	use ContainerAwareTrait;
+	use CountryCodeTrait;
+
+	/**
+	 * @var Ads
+	 */
+	protected $ads;
+
+	/**
+	 * BudgetMetricsController constructor.
+	 *
+	 * @param RESTServer $rest_server
+	 * @param Ads        $ads
+	 */
+	public function __construct( RESTServer $rest_server, Ads $ads ) {
+		parent::__construct( $rest_server );
+		$this->ads = $ads;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'ads/campaigns/budget-metrics',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_budget_metrics_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'context'       => $this->get_context_param( [ 'default' => 'view' ] ),
+			'budget'        => [
+				'description'       => __( 'Budget to fetch metrics for.', 'google-listings-and-ads' ),
+				'type'              => 'number',
+				'sanitize_callback' => $this->get_sanitize_price_callback(),
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+			'country_codes' => [
+				'description'       => __( 'List of country codes to fetch metrics for.', 'google-listings-and-ads' ),
+				'type'              => 'array',
+				'sanitize_callback' => $this->get_country_code_sanitize_callback(),
+				'validate_callback' => $this->get_country_code_validate_callback(),
+				'items'             => [
+					'type' => 'string',
+				],
+				'required'          => true,
+				'minItems'          => 1,
+			],
+		];
+	}
+
+	/**
+	 * @return callable
+	 */
+	protected function get_budget_metrics_callback(): callable {
+		return function ( Request $request ) {
+			$country_codes = $request->get_param( 'country_codes' );
+			$budget        = $request->get_param( 'budget' );
+			$currency      = $this->ads->get_ads_currency();
+			$country       = reset( $country_codes );
+
+			if ( ! $currency ) {
+				return new Response(
+					[
+						'message'       => __( 'No currency available for the Ads account.', 'google-listings-and-ads' ),
+						'budget'        => $budget,
+						'currency'      => $currency,
+						'country_codes' => $country_codes,
+					],
+					400
+				);
+			}
+
+			// Fetch metrics from the Google Ads API.
+			$budget_metrics = $this->container->get( BudgetMetrics::class );
+			$metrics        = $budget_metrics->get_metrics( $budget, $country_codes );
```

Comment:
Calling API with a negative `budget` results in 500 internal server error. Would it be better to make it return 400 bad request error instead?

![image](https://github.com/user-attachments/assets/e2e7c2d7-fd55-4bc8-910e-f376936ac2f1)

Reply:
We should be able to resolve this by using the number format: https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#numbers

If we set the minimum to 0 it should block negative numbers.

Reply:
Also good scenario to add a unit test for.

Reply:
Forgot about this comment, created a follow up PR here: https://github.com/woocommerce/google-listings-and-ads/pull/2823